### PR TITLE
feat: Expose set finalized/safe block in plugin api BlockchainService

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR description
feat: Expose set finalized/safe block in plugin api BlockchainService. This method can be used by plugins to set finalized/safe block for a POA network (such as QBFT, IBFT and Clique).

### Thanks for sending a pull request! Have you done the following?

- [x] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [x] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [x] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [x] unit tests: `./gradlew build`
- [x] acceptance tests: `./gradlew acceptanceTest`
- [x] integration tests: `./gradlew integrationTest`
- [x] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

**Review by fab-10 — APPROVED** [review]
Pretty straightforward, just thinking if we need to support some kind of checks, and allow to set these values only if we are not running a PoS network?

File: besu/src/main/java/org/hyperledger/besu/services/BlockchainServiceImpl.java [hunk 1]
```diff
@@ -135,6 +142,17 @@ public Optional<Hash> getFinalizedBlock() {
     return blockchain.getFinalized();
   }
 
+  @Override
+  public void setFinalizedBlock(final Hash blockHash) {
+    if (genesisConfigOptionsSupplier != null) {
```
Comment (jframe) [inline besu/src/main/java/org/hyperledger/besu/services/BlockchainServiceImpl.java:147]: Just checking the genesis isn't going to work for chains that transition to POS.

You could use the protocolSchedule to check the chain is POS for given blockHash by doing something like
```protocolSchedule.getByBlockHeader(blockchain.getBlockHeader(blockHash).get()).isPoS()```

**Review by fab-10 — APPROVED** [review]

**Review by jframe — APPROVED** [review]

File: plugin-api/src/main/java/org/hyperledger/besu/plugin/services/BlockchainService.java [hunk 2]
```diff
@@ -87,4 +87,12 @@ public interface BlockchainService extends BesuService {
...
    */
   Optional<Hash> getFinalizedBlock();
+
+  /**
+   * Set the finalized block for POA networks
+   *
+   * @param blockHash Hash of the finalized block
+   * @throws UnsupportedOperationException if the network is not a POA network
+   */
+  void setFinalizedBlock(Hash blockHash);
```
[the remark this answers is not in the document]
Comment (usmansaleem) [reply plugin-api/src/main/java/org/hyperledger/besu/plugin/services/BlockchainService.java]: done.

File: besu/src/main/java/org/hyperledger/besu/services/BlockchainServiceImpl.java (hunk 1 above)
Comment (usmansaleem) [reply besu/src/main/java/org/hyperledger/besu/services/BlockchainServiceImpl.java]: done.

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/plugins/BlockchainServiceFinalizedBlockPluginTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    minerNode = besu.createMinerNode("minerNode");
+    pluginNode =
+        besu.createPluginsNode("node1", List.of("testPlugins"), List.of("--rpc-http-api=UPDATER"));
+    cluster.start(minerNode, pluginNode);
+    client = new OkHttpClient();
+  }
+
+  @Test
+  @DisplayName("Calling updateFinalizedBlockV1 will set finalized block")
+  public void canUpdateFinalizedBlock() throws IOException {
```
Comment (jframe) [inline acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/plugins/BlockchainServiceFinalizedBlockPluginTest.java:56]: Should also test the updateSafeBlock method

Comment (usmansaleem) [reply acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/plugins/BlockchainServiceFinalizedBlockPluginTest.java:56]: done.

**Review by fab-10 — APPROVED** [review]

**Pull request merged**